profiler/include/profiler/profile_contraction_impl.hpp

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iomanip>
+#include <iostream>
+#include <typeinfo>
+#include <limits>
+#include <vector>
+
+#include "ck/ck.hpp"
+#include "ck/tensor_operation/gpu/device/tensor_layout.hpp"
+#include "ck/tensor_operation/gpu/device/device_contraction_multiple_d.hpp"
+#include "ck/tensor_operation/gpu/element/element_wise_operation.hpp"
+
+#include "ck/library/tensor_operation_instance/gpu/contraction_bilinear.hpp"
+#include "ck/library/tensor_operation_instance/gpu/contraction_scale.hpp"
+
+#include "ck/library/utility/check_err.hpp"
+#include "ck/library/utility/device_memory.hpp"
+#include "ck/library/utility/host_tensor.hpp"
+#include "ck/library/utility/host_tensor_generator.hpp"
+#include "ck/library/utility/literals.hpp"
+
+#include "ck/host_utility/io.hpp"
+#include "ck/library/reference_tensor_operation/cpu/reference_contraction.hpp"
+
+namespace ck {
+namespace profiler {
+
+template <typename ALayout,
+          typename BLayout,
+          typename CDLayout,
+          typename DataType,
+          typename DTupleDataType,
+          typename CDElementOp>
+int profile_contraction_impl(int do_verification,
```

Comment:
I think it would make it easier if `profiler_contraction` will have all elementwise operations as `PassThrough`. If you would like to profile contraction fused with other operation then separate profiler file should be created. Take a look at gemm profiler.

Reply:
Do we have instances for contraction with PassThrough?

Reply:
I splited in into profile_contration_scale and profile_contration_bilinear